scripts/verify-jenkins.sh

```diff
@@ -0,0 +1,230 @@
+		exit 1
+	fi
+}
+
+# The base url for the files that we need from the code repository
+GITHUB_JENKINS_COMMIT_URL="$GITHUB_JENKINS_BASE_URL/$SHA"
+
+# verify_file()
+#
+# verifies that $local_file_path/$file_name exists and that the contents match
+# those of $remote_file_path/$file_name at the provided SHA
+#
+# Example: verify_file "file.txt" "/my/local/path" "/my/remote/path"
+#
+# @param file_name; The file name to use
+# @param local_file_path; The path to the file in the container
+# @param remote_file_path; The path to the file in the github repository
+#
+# @return void
+verify_file(){
+	local file_name=$1
+	local local_file_path="${2}/${file_name}"
+	local remote_file_path="${GITHUB_JENKINS_COMMIT_URL}/${3}/${file_name}"
+
+	# The temporary file where the contents of the remote file are stored
+	local tmp_file_path="/tmp/${file_name}"
+
+	printf "[VERIFYING] %s\n" "${local_file_path}"
+
+	# Download the remote file to a temporary file, verifying the http_status code for the request
+	printf "     - downloading %s ... " "${remote_file_path}"
+	http_status_code=$(curl -s -o "${tmp_file_path}" -w "%{http_code}\n" "${remote_file_path}" 2> /dev/null)
+
+	if [ "$http_status_code" == 200 ]; then
+		printf "completed\n"
+	else
+		printf "failed\n"
+		printf "failed to download %s, http status code: %s" "${remote_file_path}" "${http_status_code}" >> $FAILED_LOG_LOCATION
+	fi
+
+	# Check if the local file exists
+	printf "     - checking if %s exists ... " "${local_file_path}"
+	if test -f "${local_file_path}}"; then
+		printf "yes\n"
+	else
+		printf "no\n"
+		echo "${local_file_path} does not exist but should" >> $FAILED_LOG_LOCATION
+	fi
+
+	# Check if the contents of the local file match those of the remote file
+	printf "     - checking if %s matches %s ... " "${local_file_path}" "${remote_file_path}"
+	if cmp -s "${local_file_path}" "${tmp_file_path}"; then
+		printf "yes\n"
+	else
+		printf "no\n"
+		printf "jenkins-version.txt comparison failed\n" >> $FAILED_LOG_LOCATION
+		printf "%s\n" "$(diff "${local_file_path}" "${tmp_file_path}")" >> $FAILED_LOG_LOCATION
+	fi
+}
+
+# verify_plugins()
+#
+# verifies that the plugins from the bundle-plugins.txt file exist and that they
+# are the at version specified by the bundle-plugins.txt file in the code repository
+# at the specified SHA
+#
+verify_plugins() {
+	while IFS= read -r plugin
+	do
+		if [[ "$plugin" =~ "Generated" ]]; then
+			continue
+		fi
+		name="${plugin%%:*}"
+		version="${plugin##*:}"
+
+		printf "Verifying %s/%s.jpi\n" "${PLUGINS_DIR}" "${name}"
+		printf "     - checking if ${PLUGINS_DIR}/%s.jpi exists ... " "$name"
+		if test -f "${PLUGINS_DIR}/${name}.jpi"; then
+			printf "yes\n"
+		else
+			printf "no\n"
+			echo "${PLUGINS_DIR}/${name}.jpi does not exist but should" >> $FAILED_LOG_LOCATION
+		fi
+
+		found_version=$(unzip -p "${PLUGINS_DIR}/${name}.jpi" META-INF/MANIFEST.MF | grep "^Plugin-Version: " | sed -e 's#^Plugin-Version: ##')
```

Comment:
A bit confused w.r.t `plugins check`, as the plugin is reported to be "not existing", but the versions are reported to be expected.
example:

> Verifying /opt/openshift/plugins/ace-editor.jpi
>      - checking if /opt/openshift/plugins/ace-editor.jpi exists ... no
>      - checking if version matches 1.1 ... yes
> Verifying /opt/openshift/plugins/ant.jpi
>      - checking if /opt/openshift/plugins/ant.jpi exists ... no
>      - checking if version matches 1.11 ... yes

Let me know, in case I have misinterpreted anything..

Reply:
Fixed